packages/usa-header/src/index.js

```diff
@@ -78,15 +79,51 @@ const toggleNonNavItems = (active) => {
   }
 };
 
+const animateClose = (el) => {
+  el.classList.add(CLOSING_CLASS);
+  const cleanup = () => {
+    if (el.classList.contains(CLOSING_CLASS)) {
+      el.classList.remove(CLOSING_CLASS);
+      el.classList.remove(VISIBLE_CLASS);
+    }
+  };
+
+  const events = [
+    "transitionend",
+    "animationend",
+    // Just in case transition/animation gets cancelled
+    "transitioncancel",
+    "animationcancel",
+  ];
+  const callback = () => {
+    console.log('callback')
+    cleanup();
+    events.forEach((e) => el.removeEventListener(e, callback));
+  };
+  events.forEach((e) => el.addEventListener(e, callback));
+};
+
 const toggleNav = (active) => {
   const { body } = document;
   const safeActive = typeof active === "boolean" ? active : !isActive();
 
   body.classList.toggle(ACTIVE_CLASS, safeActive);
 
-  select(TOGGLES).forEach((el) =>
-    el.classList.toggle(VISIBLE_CLASS, safeActive)
-  );
+  select(TOGGLES).forEach((el) => {
+    // If there's an animation/transition duration and the nav
+    // is closing, add CLOSING_CLASS for the duration instead of
+    // immediately closing. This lets the nav animate close
+    const computedStyle = getComputedStyle(el);
+    const duration =
+      parseFloat(computedStyle.animationDuration, 10) ||
+      parseFloat(computedStyle.transitionDuration);
+    if (duration !== 0 && !safeActive) {
+      animateClose(el);
+    } else {
+      el.classList.toggle(VISIBLE_CLASS, safeActive);
+      el.classList.toggle(CLOSING_CLASS, false);
+    }
+  });
```

Comment:
Could we simplify this now that we're listening for animation and transition end?

```diff
  select(TOGGLES).forEach((el) => {
-    // If there's an animation/transition duration and the nav
-    // is closing, add CLOSING_CLASS for the duration instead of
-    // immediately closing. This lets the nav animate close
-    const computedStyle = getComputedStyle(el);
-    const duration =
-      parseFloat(computedStyle.animationDuration, 10) ||
-      parseFloat(computedStyle.transitionDuration);
-    if (duration !== 0 && !safeActive) {
+    if (!safeActive) {
      animateClose(el);
    } else {
      el.classList.toggle(VISIBLE_CLASS, safeActive);
    }
  });
```

Reply:
Not necessarily. If the animation/transition doesn't run at all, then the events won't fire. This can happen:
* If the consuming developer overrides the animation to prevent it from ever triggering
* If the CSS fails to load
* **If the user has `prefers-reduced-motion: reduce` enabled in their browser**

Obviously the first 2 are edge cases, but the 3rd is a situation which we should expect to happen. From the [docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/transitionend_event):
> If there is no transition delay or duration, **if both are 0s or neither is declared**, there is no transition, and **none of the transition events are fired**.

So we need to account for the case where the `animationDuration` or `transitionDuration` is `0s` and none of the 4 end/cancel events ever fire. We could listen for `animationstart`/`transitionstart` and if we don't hear it within Xms, then remove the transition class. But that seems less elegant (and less synchronous) than checking if the duration is 0 at all.
If we're only concerned with the 3rd bullet above, then I could possibly add `transition-delay: 1ms` and `animation-delay: 1ms`. Even if the duration is 0, I believe the events will still fire. Let me know if you'd like me to explore that.

Reply:
Added comment explaining this in the code.